src/RA_AchievementOverlay.cpp

```diff
@@ -1583,41 +1584,38 @@ void AchievementOverlay::InstallNewsArticlesFromFile()
 {
     m_LatestNews.clear();
 
-    FILE* pf = nullptr;
-    fopen_s(&pf, RA_NEWS_FILENAME, "rb");
-    if (pf != nullptr)
-    {
-        Document doc;
-        doc.ParseStream(FileStream(pf));
-
-        if (doc.HasMember("Success") && doc["Success"].GetBool())
-        {
-            const Value& News = doc["News"];
-            for (SizeType i = 0; i < News.Size(); ++i)
-            {
-                const Value& NextNewsArticle = News[i];
-
-                NewsItem nNewsItem;
-                nNewsItem.m_nID = NextNewsArticle["ID"].GetUint();
-                nNewsItem.m_sTitle = NextNewsArticle["Title"].GetString();
-                nNewsItem.m_sPayload = NextNewsArticle["Payload"].GetString();
-                nNewsItem.m_sAuthor = NextNewsArticle["Author"].GetString();
-                nNewsItem.m_sLink = NextNewsArticle["Link"].GetString();
-                nNewsItem.m_sImage = NextNewsArticle["Image"].GetString();
-                nNewsItem.m_nPostedAt = NextNewsArticle["TimePosted"].GetUint();
 
-                tm destTime;
-                localtime_s(&destTime, &nNewsItem.m_nPostedAt);
+    std::ifstream ifile{ RA_NEWS_FILENAME, std::ios::binary };
 
-                char buffer[256];
-                strftime(buffer, 256, "%b %d", &destTime);
-                nNewsItem.m_sPostedAt = buffer;
+    Document doc;
+    IStreamWrapper isw{ ifile };
+    doc.ParseStream(isw);
 
-                m_LatestNews.push_back(nNewsItem);
-            }
+    if (doc.HasMember("Success") && doc["Success"].GetBool())
+    {
+        const auto& News = doc["News"];
+        for (auto& i : News.GetArray())
+        {
+            NewsItem nNewsItem{
+                i["ID"].GetUint(),
+                i["Title"].GetString(),
+                i["Payload"].GetString(),
+                i["TimePosted"].GetUint(),
+                std::to_string(i["TimePosted"].GetUint()), // does the format really matter?
+                i["Author"].GetString(),
+                i["Link"].GetString(),
+                i["Image"].GetString()
+            };
+
+            //tm destTime;
```

Comment:
Commented out just in case the format does matter, the `to_string `gives the same value as the json response. The timestamp that is.